<launch>

  <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch">
      <arg name="align_depth"         value="true"/>        
      <arg name="enable_sync"         value="true"/>        
      <arg name="color_width"         value="640"/>
      <arg name="color_height"        value="480"/>
      <arg name="depth_width"         value="640"/>
      <arg name="depth_height"        value="480"/>
      <arg name="infra_width"         value="640"/>
      <arg name="infra_height"        value="480"/>
      <arg name="color_fps"           value="30"/>          
      <arg name="depth_fps"           value="30"/>          
      <arg name="infra_fps"           value="30"/>          
      <arg name="enable_gyro"         value="false"/>       
      <arg name="enable_accel"        value="false"/>       
      <arg name="enable_pointcloud"   value="false"/>       
      <arg name="enable_infra1"       value="true"/>       
      <arg name="enable_infra2"       value="true"/>       
      <arg name="allow_no_texture_points" value="true"/>
  </include> -->

   <include file="$(find azure_kinect_ros_driver)/launch/azure_kinect_driver.launch">
    <arg name="color_resolution" value="720P"/>
    <arg name="depth_mode"       value="NFOV_UNBINNED"/>
    <arg name="camera_fps"       value="30"/>
  </include> 
 
  <node pkg="yolo_realsense_kinect" type="yolo_kinect_loop_node" name="yolo_kinect_loop_node" output="screen">
      <param name="engine_name" value="$(find yolo_realsense_kinect)/weight/yolo11n_loop_8.3.0_6.5_.engine"/> 
      <param name="cuda_post_process" value="g"/>

      <param name="conf_thresh" value="0.55"/>
      <param name="nms_thresh" value="0.45"/>

   

      <param name="depth_topic" value="/k4a/aligned_depth_to_color/image_raw"/>
      <param name="info_topic"  value="/k4a/color/camera_info"/>

      <param name="loop_distance_max" value="5.0" />
      <param name="loop_distance_min" value="1.2" />
  </node> 
  
  <node pkg="yolo_realsense_kinect" type="yolo_kinect_circle_node" name="yolo_kinect_circle_node" output="screen">
      <param name="engine_name" value="$(find yolo_realsense_kinect)/weight/yolo11n_circle_8.3.0_6.5_.engine"/> 
      <param name="cuda_post_process" value="g"/>

      <param name="conf_thresh" value="0.55"/>
      <param name="nms_thresh" value="0.45"/>

      <param name="color_topic" value="/k4a/color/image_raw"/>
      <param name="depth_topic" value="/k4a/aligned_depth_to_color/image_raw"/>
      <param name="info_topic"  value="/k4a/color/camera_info"/>

      <param name="circle_distance_max" value="3.0" />
      <param name="circle_distance_min" value="0.2" />
  </node>

  </launch>